contrib/globalsign/mgo/mgo_test.go

```diff
@@ -0,0 +1,355 @@
+package mgo
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/globalsign/mgo"
+	"github.com/globalsign/mgo/bson"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+func TestMain(m *testing.M) {
+	_, ok := os.LookupEnv("INTEGRATION")
+	if !ok {
+		fmt.Println("--- SKIP: to enable integration test, set the INTEGRATION environment variable")
+		os.Exit(0)
+	}
+	os.Exit(m.Run())
+}
+
+func testMongoCollectionCommand(assert *assert.Assertions, command func(*Collection)) []mocktracer.Span {
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	parentSpan, ctx := tracer.StartSpanFromContext(
+		context.Background(),
+		"mgo-unittest",
+		tracer.SpanType("app"),
+		tracer.ResourceName("insert-test"),
+	)
+
+	session, err := Dial("localhost:27017", WithServiceName("unit-tests"), WithContext(ctx))
+	defer session.Close()
+
+	assert.NotNil(session)
+	assert.Nil(err)
+
+	db := session.DB("my_db")
+	collection := db.C("MyCollection")
+
+	command(collection)
+
+	parentSpan.Finish()
+
+	spans := mt.FinishedSpans()
+	return spans
+}
+
+func TestCollection_Insert(t *testing.T) {
+	assert := assert.New(t)
+
+	entity := bson.D{
+		bson.DocElem{
+			Name: "entity",
+			Value: bson.DocElem{
+				Name:  "index",
+				Value: 0}}}
+
+	insert := func(collection *Collection) {
+		collection.Insert(entity)
+	}
+
+	spans := testMongoCollectionCommand(assert, insert)
+	assert.Equal(2, len(spans))
+	assert.Equal("mongodb.query", spans[0].OperationName())
+}
+
+func TestCollection_Update(t *testing.T) {
+	assert := assert.New(t)
+
+	entity := bson.D{
+		bson.DocElem{
+			Name: "entity",
+			Value: bson.DocElem{
+				Name:  "index",
+				Value: 0}}}
+
+	insert := func(collection *Collection) {
+		collection.Insert(entity)
+		collection.Update(entity, entity)
+	}
+
+	spans := testMongoCollectionCommand(assert, insert)
+	assert.Equal(3, len(spans))
+	assert.Equal("mongodb.query", spans[1].OperationName())
+}
+
+func TestCollection_UpdateId(t *testing.T) {
+	assert := assert.New(t)
+
+	entity := bson.D{
+		bson.DocElem{
+			Name: "entity",
+			Value: bson.DocElem{
+				Name:  "index",
+				Value: 0}}}
+
+	insert := func(collection *Collection) {
+		collection.Insert(entity)
+		var r bson.D
+		collection.Find(entity).Iter().Next(&r)
+		collection.UpdateId(r.Map()["_id"], entity)
+	}
+
+	spans := testMongoCollectionCommand(assert, insert)
+	assert.Equal(5, len(spans))
+	assert.Equal("mongodb.query", spans[3].OperationName())
+}
+
+func TestCollection_Upsert(t *testing.T) {
+	assert := assert.New(t)
+
+	entity := bson.D{
```

Comment:
For example: this is copy/pasted everywhere. Just make one (global) var called `bsonEntity` or something.